### PR TITLE
Fix Warnings

### PR DESCRIPTION
**Pull request by rhyshort**
## What 
Fix numerous compilation warnings.

## How

- Correct String formatting args by casting directly to an int.
- Remove unused TDSocketChangeTracker.h
- Remove MYBlockUtils and use `performSelector`at call sites.
- Ignore some warnings which can be safely ignored using `#pragma` directives

## Reviewers

reviewer: @brynh

### CODE REVIEW

File: CDTDatastore/Encryption/CDTEncryptionKey.m [hunk 1]
```diff
@@ -32,7 +32,13 @@ - (NSData *)data {
 }
 
 #pragma mark - Init object
-- (instancetype)init { return [self initWithData:nil]; }
+- (instancetype)init
+{
+#pragma clang diagnostic push
```
Comment (brynh) [inline CDTDatastore/Encryption/CDTEncryptionKey.m]: Wouldn't it be easier and clearer to just implement this as `return nil;` as it's marked unavailable?

File: CDTDatastore/Encryption/CDTEncryptionKeySimpleProvider.m [hunk 2]
```diff
@@ -26,7 +26,13 @@ @interface CDTEncryptionKeySimpleProvider ()
 
 @implementation CDTEncryptionKeySimpleProvider
 
-- (instancetype)init { return [self initWithKey:nil]; }
+- (instancetype)init
+{
+#pragma clang diagnostic push
```
Comment (brynh) [inline CDTDatastore/Encryption/CDTEncryptionKeySimpleProvider.m]: Wouldn't it be easier and clearer to just implement this as `return nil;` as it's marked unavailable?

File: CDTDatastore/HTTP/CDTURLSessionTask.m [hunk 3]
```diff
@@ -203,15 +202,19 @@ - (void) completedThread:(NSThread *)thread {
...
+                                  onThread:thread
+                                withObject:self.requestError
+                             waitUntilDone:NO];
         } else {
-            MYOnThread(thread, ^{
-                [self.delegate receivedResponse:self.response];
-                [self.delegate receivedData:self.requestData];
-            });
-            
+            [self.delegate performSelector:@selector(receivedResponse:)
```
Comment (brynh) [inline CDTDatastore/HTTP/CDTURLSessionTask.m]: The behaviour of this is now changed by waiting on the current thread for the `receivedResponse:` to complete and then calling `receivedData` whereas previously they were both called without waiting on the current thread. I think perhaps the method calls:

> [self.delegate receivedResponse:self.response];
> [self.delegate receivedData:self.requestData];

should be put in a separate wrapper method and then called with a single call to `performSelector` that calls the wrapper method on the separate thread and doesn't wait until done.

File: CDTDatastore/Encryption/CDTEncryptionKey.m (hunk 1 above)
Comment (rhyshort) [inline CDTDatastore/Encryption/CDTEncryptionKey.m]: done in d7b1e3f

File: CDTDatastore/Encryption/CDTEncryptionKeySimpleProvider.m (hunk 2 above)
Comment (rhyshort) [inline CDTDatastore/Encryption/CDTEncryptionKeySimpleProvider.m]: done in d7b1e3f

File: CDTDatastore/HTTP/CDTURLSessionTask.m (hunk 3 above)
Comment (rhyshort) [inline CDTDatastore/HTTP/CDTURLSessionTask.m]: done in d7b1e3f

**Comment by brynh** [comment]
LGTM.  +1

**Pull request merged**